force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,168 @@
+    }
+    
+
+
+    // Test method for handleAutoLeadScoring
+    @isTest
+    static void testAutoLeadScoring() {
+        // Create test leads with different scoring criteria
+        List<Lead> testLeads = new List<Lead>{
+            new Lead(
+                FirstName='Test1',
+                LastName='Lead1',
+                Company='Test Co',
+                LeadSource='Website',
+                Email='test1@example.com',
+                Phone='1234567890',
+                Industry='Technology'
+            ),
+            new Lead(
+                FirstName='Test2',
+                LastName='Lead2',
+                Company='Test Co',
+                LeadSource='Website',
+                Email='test2@example.com'
+            ),
+            new Lead(
+                FirstName='Test3',
+                LastName='Lead3',
+                Company='Test Co',
+                Phone='9876543210'
+            )
+        };
+        
+        Test.startTest();
+        LeadTriggerHandler.handleAutoLeadScoring(testLeads);
+        Test.stopTest();
+        
+        // Verify scores
+        System.assertEquals(18, testLeads[0].Lead_Score__c, 'Lead should have maximum score of 18');
+        System.assertEquals(3, testLeads[1].Lead_Score__c, 'Lead should have score of 3 for website source and email');
+        System.assertEquals(5, testLeads[2].Lead_Score__c, 'Lead should have score of 5 for phone');
+    }
+ 
+// Test method for handleLeadAutoConvert (positive case)
+ @isTest
+ static void testHandleLeadAutoConvert(){
+    List <Lead> leadList = new List<Lead>();
+    Lead  l1 = new Lead(LastName = 'Test1234', Company = 'Test Comp', Email = 'TL1@TEST.COM');
+    Lead  l2 = new Lead(LastName = 'Test123', Company = 'Test Comp', Email = 'L2@TEST.COM');
+    leadList.add(l1);
+    leadList.add(l2);
+    //insert leadList;
+
+     Test.startTest();
+     insert leadList;
+     //LeadTriggerHandler.handleLeadAutoConvert(leadList);
+     Test.stopTest();
+    //  Database.LeadConvert lc = new Database.LeadConvert();
+    //      lc.setLeadId(l.Id);
+   
+    Lead convertedLead = [SELECT Id, IsConverted FROM Lead WHERE Id = :l1.Id];
+     System.assertEquals(true, convertedLead.IsConverted, 'Lead should have been converted');
+
+    
+ }
+ 
+
+ // Test method for handleLeadAutoConvert (negative case - no matching contact)
+ @IsTest
+ static void testHandleLeadAutoConvertNegative(){
+    List <Lead> leadList = new List<Lead>();
+    Lead  lead2 = new Lead(LastName = 'Testlead123', Company = 'Test Comp', Email = 'L2@TEST.COM');
+    leadList.add(lead2);
+    //insert leadList;
+
+     Test.startTest();
+     insert leadList;
+     //LeadTriggerHandler.handleLeadAutoConvert(leadList);
+     Test.stopTest();
+     Lead convertedLead = [SELECT Id, IsConverted FROM Lead WHERE Id = :lead2.Id];
+     System.assertEquals(false, convertedLead.IsConverted, 'Lead should not have been converted');
```

Comment:
Try to see if you can combine this negative test case with the positive one above lead2 looks very similar to l2 above